### ISSUE TITLE
Apparently installs properly on Mac OS X Catalina, using sbcl 2.1.5: but nothing runs

### ISSUE THREAD

**Issue by m-bob** [phantomics/april#184]
Hi,

I followed the quicklisp install instructions for april, which went exactly as the documentation states. Specifically, I get

* (ql:quickload 'April)
To load "april":
  Load 1 ASDF system:
    april
; Loading "april"
....
Specified idiom ｢APRIL｣ with 7 opening grammar patterns, 10 following grammar patterns, 135 lexical functions, 15 lexical operators, 18 utility functions and 848 unit tests.

("april")

There's no indication of a problem, but I'm unable to execute any april  command. (april  (demo)), (april "x←5 ⋄ $[x>3;8;12]"), etc. all fail to execute. The error message is always basically the same:  undefined function:  COMMON-LISP-USER::APRIL, undefined function:  COMMON-LISP-USER::APRIL-F .

I'm not an experienced lisp, sbcl, quicklisp  user, so perhaps I'm missing something basic. (I hope.)

Can anyone help?

Thanks!

Murray

**Comment by phantomics** [comment]
Hi Murray, those April functions are located inside the `APRIL` software package and you're entering your commands from the `CL-USER` package. Solution 1 is to run the commands like this:

```lisp
* (april:april (demo))

* (april:april "x←5 ⋄ $[x>3;8;12]")
```

Putting `april:` on the front tells the Lisp instance to look for the functions inside the `APRIL` package.

Solution 2 is to change the package you're in to April. To do that, enter, `(in-package :april)`. Then you can enter the commands as you did above.

If you're using Slime, you can also enter the April package by going to the REPL and typing :
`,`
`!p` (and hit Enter)
`april` (and hit Enter)

Remember that to exit the Lisp session you need to use the `(quit)` command which is in the `CL-USER` package, so running it from inside the April package requires that you enter `(cl-user::quit)`. Hope this helps.

**Comment by m-bob** [comment]
Thanks so much Andrew!

That's exactly what I needed to know and everything works great!  I guess I
should be embarrassed that I didn't know that --but I'm really not. And I'm
not sure if or how I was ever going to figure it out -- I know we're
"supposed" to read the documentation, but there's just so much and it's a
challenge (for me) just to get the basics of the language down.

You saved me an enormous amount of time -- thank you!

Also, I've long thought that integration of APL and Lisp would be a great
thing to explore -- so thank you very much for creating April! I'm looking
forward to playing with it.

Again, thanks!

Murray

**Comment by phantomics** [comment]
You're welcome Murray. Here are some videos that can help you get started with a "proper" Lisp UX:

https://www.youtube.com/watch?v=_B_4vhsmRRI
https://www.youtube.com/watch?v=NUpAvqa5hQw

This is a good way to get acquainted with Lisp's REPL features. Are you using SBCL from the command line, just by typing "sbcl"? You can do this but it's very clunky. I take it you have prior APL experience and you're now getting into Lisp? Lisp's REPL environment is quite different from APL's; most users prefer to interact with it through the Emacs editor and its Slime plugin, as shown in the videos above, which offer many quality-of-life improvements over a simple CLI. It's a more modular system than APL interpreters that have their own self-contained console environments.

Let me know how your exploration of April goes and if you have any more questions. Do you know yet what you'd like to use it for? Best,

Andrew

**Comment by m-bob** [comment]
Thank you. I'm more a dabbler in APL than anything else -- but I've dabbled
a long time.

Thanks for the videos. I've been using sbcl from the command line and also
through roswell, which is much more pleasant. I've tried to get comfortable
with emacs, but it's never happened.

At the moment, I have no clear end goal. A lot of my interest in different
computer languages is how they affect how I approach problems; not
necessarily that I'll end up using them specifically, but more how they
expand my thinking.  I think Lisp and APL (and Haskell, which I've found
particularly challenging) are "mind-expanding" and provide different ways
of looking at computation and even mathematics.
